.eslintrc.js

```diff
@@ -0,0 +1,125 @@
+        'jest/valid-expect': 'error',
+        'jest/no-commented-out-tests': 'off',
+      },
+    },
+  ],
+  extends: [
+    'eslint:recommended',
+    'plugin:react/recommended',
+    'plugin:jest/recommended',
+  ],
+  parser: 'babel-eslint',
+  parserOptions: {
+    ecmaFeatures: {
+      jsx: true,
+    },
+    ecmaVersion: 12,
+    sourceType: 'module',
+    project: './jsconfig.json',
+  },
+  plugins: [
+    'import',
+    'react',
+    'jest',
+  ],
+  rules: {
+    'linebreak-style': ['error', 'unix'],
+
+    'no-unused-vars': [
+      'error',
+      { argsIgnorePattern: '^_' },
+    ],
+
+    // 2 space indentation
+    'indent': ['error', 2],
+
+    // Style
+    'quotes': ['error', 'single', { avoidEscape: true }],
+
+    // ensures clean diffs, see https://medium.com/@nikgraf/why-you-should-enforce-dangling-commas-for-multiline-statements-d034c98e36f8
+    'comma-dangle': ['error', 'always-multiline'],
+
+    // Require all imported dependencies are actually declared in package.json
+    'import/no-extraneous-dependencies': [
+      'error',
+      {
+        optionalDependencies: false,    // Disallow importing optional dependencies (those shouldn't be in use in the project)
+        peerDependencies: false,         // Disallow importing peer dependencies (that aren't also direct dependencies)
+      },
+    ],
+
+    // Require all imported libraries actually resolve (!!required for import/no-extraneous-dependencies to work!!)
+    'import/no-unresolved': ['error'],
+
+    // Require an ordering on all imports
+    'import/order': ['warn', {
+      groups: ['builtin', 'external'],
+      alphabetize: { order: 'asc', caseInsensitive: true },
+    }],
+
+    // Cannot import from the same module twice
+    'no-duplicate-imports': ['error'],
+
+    // Cannot shadow names
+    'no-shadow': ['error'],
+
+    // Required spacing in property declarations (copied from TSLint, defaults are good)
+    'key-spacing': ['error'],
+
+    // Require semicolons
+    'semi': ['error', 'always'],
+
+    // Don't unnecessarily quote properties
+    'quote-props': ['error', 'consistent-as-needed'],
+
+    // No multiple empty lines
+    'no-multiple-empty-lines': ['error'],
+
+    // Max line lengths
+    'max-len': ['error', {
+      code: 120,
+      ignoreUrls: true, // Most common reason to disable it
+      ignoreStrings: true, // These are not fantastic but necessary for error messages
+      ignoreTemplateLiterals: true,
+      ignoreComments: true,
+      ignoreRegExpLiterals: true,
+    }],
+
+    // Useless diff results
+    'no-trailing-spaces': ['error'],
+
+    // Must use foo.bar instead of foo['bar'] if possible
+    'dot-notation': ['error'],
+
+    // Are you sure | is not a typo for || ?
+    'no-bitwise': ['error'],
+
+    // Not enforcing props validation for React
+    'react/prop-types': 'off',
+
+    // Don't leave log statements littering the premises!
+    // 'no-console': ['error'],
+  },
+};
```

Comment:
Trailing space (minor)

Reply:
do you mean add an enter?

Reply:
Hey there, I'm not Timothy but that's right! Trailing space is a minor thing, but it helps with file versioning.